### ISSUE TITLE
sgx-detect: does not understand EPC information on ice lake

### ISSUE THREAD

**Issue by aandyl** [fortanix/rust-sgx#332]
On an ice lake machine:

```
$ ./sgx-detect  --verbose
Detecting SGX, this may take a minute...
？ SGX instruction set
  ✔  CPU support
  ✔  CPU configuration
  ✔  Enclave attributes
  ？ Enclave Page Cache
  SGX features
    ✔  SGX2  ✔  EXINFO  ✘  ENCLV  ✘  OVERSUB  ✔  KSS  
    Total EPC size: 0.0MiB
✔  Flexible launch control
  ✔  CPU support
  ？ CPU configuration
  ✔  Able to launch production mode enclave
✔  SGX system software
  ✔  SGX kernel device (/dev/sgx/enclave)
  ✔  libsgx_enclave_common
  ✔  AESM service
  ✔  Able to launch enclaves
    ✔  Debug mode
    ✔  Production mode
    ✔  Production mode (Intel whitelisted)

🕮  SGX instruction set > Enclave Page Cache
Your hardware supports SGX, but the Enclave Page Cache (EPC) is misconfigured. This could indicate a CPU or hypervisor bug.

debug: CPUID.(EAX=12H,ECX=2H):ECX[3:0] = Unknown
debug: CPUID.(EAX=12H,ECX=3H):ECX[3:0] = Unknown

More information: https://edp.fortanix.com/docs/installation/help/#invalid-cpuid
```

**Comment by jethrogb** [comment]
Ah yes we need to understand the new EPC type 0x2.

**Comment by jethrogb** [comment]
Why is the warning at line https://github.com/fortanix/rust-sgx/blob/master/sgxs-tools/src/sgx_detect/interpret.rs#L150 not printed?

**Comment by aandyl** [comment]
With `RUST_LOG=debug` I get:

```
[2021-04-07T15:20:20Z WARN  sgx_detect::interpret] CPUID 12h, sub-leaf 0 EAX has reserved bits set: 00000400
[2021-04-07T15:20:20Z WARN  sgx_detect::interpret] CPUID 12h, sub-leaf 2 (EPC section) unknown EPC type: 2h. EAX=18000001, EBX=00000030, ECX=e7800002, EDX=0000000f
[2021-04-07T15:20:20Z WARN  sgx_detect::interpret] CPUID 12h, sub-leaf 3 (EPC section) unknown EPC type: 2h. EAX=18000001, EBX=00000070, ECX=e8000002, EDX=0000000f
```

**Comment by 6open** [comment]
I wonder if this problem has been solved

**Comment by raoulstrackx** [comment]
Yes it was fixed in #339
